Sources/Turf/WKT.swift

```diff
@@ -0,0 +1,222 @@
+            guard let object = (data as? [Point])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'Point' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension MultiPoint: WKTCodable {
+    public var WKTString: String {
+        return "MULTIPOINT\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [MultiPoint])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'MultiPoint' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension LineString: WKTCodable {
+    public var WKTString: String {
+        return "LINESTRING\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [LineString])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'LineString' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension MultiLineString: WKTCodable {
+    public var WKTString: String {
+        return "MULTILINESTRING\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [MultiLineString])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'MultiLineString' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension Polygon: WKTCodable {
+    public var WKTString: String {
+        return "POLYGON\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [Polygon])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'Polygon' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension MultiPolygon: WKTCodable {
+    public var WKTString: String {
+        return "MULTIPOLYGON\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [MultiPolygon])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'MultiPolygon' from WKT with error: \(error)")
```

Comment:
Should there be a throwable initializer in order to pass the error further up the stack? If the error is ignorable, then perhaps we shouldn’t spew it to the console.

Reply:
Didn't want to silently fail, so put a log message. But in the end Error is preferable way, so refactored parser a bit for that purpose. Thanks for flagging it